### DISCUSSION TITLE
Keyword vs Text

### DISCUSSION THREAD

**Discussion by squares-ui** [Security-Onion-Solutions/securityonion#2594] [Unsupported Versions] [answered]
In SO2 a lot of fields are indexed into Elastic as text, out of interest, why was Keyword not chosen?

Some fields (e.g. IP, network transport) are only 1 word
Other fields (e.g. ruleset = "emerging threats") are multi word, but there is limited value in indexing each word separately? (as these words would always go together)

I ask because Text types prevent sortable and aggregatable etc.  And surely keyword is faster?  Guessing something I'm overlooking :)

Thanks

**Comment by weslambert** [comment, marked as answer]
Many of them are indexed as text AND keyword.  Is there one in particular you are having an issue with?

**Comment by squares-ui** [comment]
I See, keyword is an attribute in the mapping, rather than the type.

Excellent, makes sense, thanks Wes
